tsinfer/inference.py

```diff
@@ -1434,17 +1431,30 @@ def __init__(
             num_alleles=num_alleles, max_nodes=self.max_nodes, max_edges=self.max_edges
         )
         logger.debug(f"Allocated tree sequence builder with max_nodes={self.max_nodes}")
-        if resume_lmdb_file is not None:
-            self.resume_lmdb = lmdb.open(
-                # Windows doesn't have sparse files, so use a smaller map size
-                resume_lmdb_file,
-                subdir=False,
-                map_size=1 * 1024 * 1024 * 1024
-                if sys.platform == "win32"
-                else 100 * 1024 * 1024 * 1024,
-            )
-        else:
-            self.resume_lmdb = None
+
+        self.match_data = collections.defaultdict(dict)
+        self.match_data_dir = match_data_dir
+        if match_data_dir is not None:
+            os.makedirs(match_data_dir, exist_ok=True)
+            for file in os.listdir(match_data_dir):
+                with open(os.path.join(match_data_dir, file), "rb") as f:
+                    (group, num_sites), batch_results = pickle.load(f)
```

Comment:
It's worthwhile making dataclass for this payload, we'll inevitably need to add more things to it.